tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchTextBuilderFacts.cs

```diff
@@ -31,7 +36,42 @@ public void WhenLuceneQuery_TreatsLeadingIdAsPackageId(bool luceneQuery, string
                     LuceneQuery = luceneQuery,
                 });
 
-                Assert.Equal(expected, actual);
+                Assert.Equal(expected, actual.Text);
+            }
+
+            [Theory]
+            [InlineData("id:hello", "hello")]
+            [InlineData("packageid:hello", "hello")]
+            [InlineData("packageid:\"hello\"", "hello")]
+            [InlineData("packageid:\"HIthereFRIEND\"", "HIthereFRIEND")]
+            [InlineData("packageid:" + MaxId, MaxId)]
+            public void ExtractsPackageIdWhenThereIsOnlyOneClause(string query, string expected)
+            {
+                var actual = _target.V2Search(new V2SearchRequest
+                {
+                    Query = query,
+                    LuceneQuery = true,
+                });
+
+                Assert.Equal(expected, actual.PackageId);
+            }
+
+            [Theory]
+            [InlineData("id:hello version:1.0.0")]
+            [InlineData("packageid:foo--bar")] // Invalid package ID
+            [InlineData("packageid:" + MaxId + "a")]
+            [InlineData("packageid:hello a")]
+            [InlineData("packageid:\"hello a\"")]
+            [InlineData("hello")]
```

Comment:
Should we test if there are multiple package id terms?

Reply:
Yes. Added that and more.